consensus/istanbul/validator/default.go

```diff
@@ -237,7 +277,7 @@ func (valSet *defaultSet) RemoveValidators(removedValidators *big.Int) bool {
 func (valSet *defaultSet) Copy() istanbul.ValidatorSet {
 	valSet.validatorMu.RLock()
 	defer valSet.validatorMu.RUnlock()
-	newValSet := NewSet(MapValidatorsToData(valSet.validators))
+	newValSet := NewSetWithCache(MapValidatorsToDataWithCache(valSet.validators))
```

Comment:
do we need to change this method?? it's not used by snapshot

Reply:
It's used by Snapshot.copy(), so it's needed...